File: 2550-count-collisions-of-monkeys-on-a-polygon/2550-count-collisions-of-monkeys-on-a-polygon.py

class Solution:
    def monkeyMove(self, n: int) -> int:
        x=n
        y=2
        m=7+10**9
        k=10000
        o=1000
        str="fdjlasldfjkdsklfjasdjklfjklsdafjkljkl"
        #leetcode contest :)
        #this is not done with dp
        return (pow(y,x,m)-2)%m

File: 2550-count-collisions-of-monkeys-on-a-polygon/2550-count-collisions-of-monkeys-on-a-polygon.java
class Solution {
    long m=1000_000_007;
    public long pow(int x,int n)
    {
        if(x==1)
            return 1;
        if(n==1)
            return x;
        long p=pow(x,n/2)%m;
        if(n%2==1)
            return x%m*p%m*p%m;
        return p%m*p%m;
    }
    public int monkeyMove(int n) {
        long k=pow(2,n);
        return (int)((k-2+m)%m);
    }
}